include/boost/url/decode.hpp

```diff
@@ -74,14 +67,12 @@ namespace urls {
     @see
         @ref decode_opts.
 */
-template<
-    class CharSet =
-        grammar::all_chars_t>
-result<decode_view>
+template<BOOST_URL_STRTOK_TPARAM>
+result<BOOST_URL_STRTOK_RETURN>
 decode(
-    string_view s,
-    decode_opts const& opt = {},
-    CharSet const& allowed = {}) noexcept;
+    BOOST_URL_PCT_STRING_VIEW s,
+    BOOST_URL_STRTOK_ARG(token),
```

Comment:
Are you sure the token should come before the opts?

Reply:
Not at all :)

Done.